File: app/src/main/java/com/example/sercanyusuf/boomsetdemo/di/modules/AppModule.java
package com.example.sercanyusuf.boomsetdemo.di.modules;

import android.app.Application;
import android.content.Context;

import javax.inject.Singleton;

import dagger.Module;
import dagger.Provides;
@Module
public class AppModule {

    private Application mApp;

    public AppModule(Application app) {
        this.mApp = app;
    }

    @Provides
    @Singleton
    public Context provideContext(){
        return mApp;
    }
}

File: app/src/main/java/com/example/sercanyusuf/boomsetdemo/di/modules/AppModule.kt
package com.example.sercanyusuf.boomsetdemo.di.modules

import com.example.sercanyusuf.boomsetdemo.BoomsetApp
import com.example.sercanyusuf.boomsetdemo.data.repository.DateRepositoryImplementation
import com.example.sercanyusuf.boomsetdemo.data.repository.EventRepository
import com.example.sercanyusuf.boomsetdemoapp.di.scope.PerApplication
import dagger.Module
import dagger.Provides
import javax.inject.Singleton

@Module
class AppModule(private val app: BoomsetApp) {
    @Provides
    @Singleton
    fun provideApplication() = app

    @Provides
    @PerApplication
    internal fun provideLoginInteractor(dataManager: DateRepositoryImplementation): EventRepository {
        return dataManager
    }

}